network/transport/types.go

```diff
@@ -106,3 +96,59 @@ type ConnectorStats struct {
 
 // NutsCommServiceType holds the DID document service type that specifies the Nuts network service address of the Nuts node.
 const NutsCommServiceType = "NutsComm"
+
+// ParseNutsCommAddress parses the given input string to a gRPC target address.
+// The input must include the protocol scheme (e.g. grpc://).
+// The address must NOT be an IP address.
+// The input must not be a reserved address or TLD as described in RFC2606 or https://www.ietf.org/archive/id/draft-chapin-rfc2606bis-00.html.
+func ParseNutsCommAddress(input string) (string, error) {
+	parsed, err := url.Parse(input)
+	if err != nil {
+		return "", err
+	}
+	if parsed.Scheme != "grpc" {
+		return "", errors.New("invalid URL scheme")
+	}
+	if net.ParseIP(parsed.Hostname()) != nil {
+		return "", errors.New("hostname is IP")
+	}
+	if isReserved(parsed) {
+		return "", errors.New("hostname is reserved")
+	}
+	return parsed.Host, nil
+}
+
+// isReserved returns true if URL uses any of the reserved TLDs or addresses
+func isReserved(URL *url.URL) bool {
+	parts := strings.Split(strings.ToLower(URL.Hostname()), ".")
+	if len(parts) < 2 {
```

Comment:
this could break docker-compose setups. Maybe only disallow in strict mode?

Reply:
This is only used in NutsComm service discovery, bootstrap nodes do not check for this.
- Reserved addresses will generally not resolve to anything, and when they do (like localhost), the nodeDIDs will not match and probably cause some weird hard to debug error messages.
- Internal endpoints should never be published publicly?

Reply:
The private-transactions end-to-end test registers a Docker service+port (`nodeA:5555`) as NutsComm address in `prepare.sh`

We could always rename the node in the test, or add a network link so we can use something like `nodeA.docker` which isn't disallowed.

Reply:
E2E tests currently don't fail because the node is bootstrapped. 
For Node Discovery to work there needs to be an intermediary node that connects nodes that are not directly connected, so we need a minimum of 3 nodes. We can extend this e2e test with a third node, or remove the part that publishes the endpoint.